### PR TITLE
LL-4850 - Alert redesign

### PR DESCRIPTION
**Pull request by ghost**
### Type

UI Polish

### Tickets
EPIC: https://ledgerhq.atlassian.net/browse/LL-4441
https://ledgerhq.atlassian.net/browse/LL-4850
https://ledgerhq.atlassian.net/browse/LL-4661
https://ledgerhq.atlassian.net/browse/LL-4664

### Context

The new Design System is being polished, and some existing components having same/similar styles should be refactored to use the design system components.

We start with the Alert component that will replace many existing components.
It inherits mainly from the `InfoBox` component.

Figma: https://www.figma.com/file/f3Q65y0n7sCnN9PsCMrhz6/Live-Library?node-id=302%3A42

### Specs

image

Alert is the new component for giving semantic feedback in the form of a colored Box, with a semantic icon on the left, an optional bold title, a text (single line only), and a right component, all vertically centered with margin inbetween.

Props:
- `title`: Optional title for the first line of alert (semi-bold text)
- `left`: replace the icon by anything (used to replace the icon if needed)
- `right`: display something on the right (e.g. for including a button) 
- `noIcon`: do not display the icon
- `children`: the alert content

We can display a "learn more" link at the end of the content (same line):
- `onLearnMore`: callback when click on learnMore
- `learnMoreUrl`: alternative to using callback (onLearnMore still take precedence)
- `learnMoreLabel`: customize learnMore label (defaults to translation `common.learnMore`)
- `learnMoreIsInternal` - boolean: remove the external icon - defaults: false
- `learnMoreOnRight` - boolean: display the learn more link on the right (right still take precedence)

It has a limited list of variants:
* `primary`: info circle icon - blue
* `secondary`: info circle icon - grey (NOTE: used as *default*)
* `success`: check circle icon - green
* `warning`: exclamation circle icon - orange
* `error`: cross circle icon - red
* `hint`: new LightBulb icon - blue
* `security`: red shield big icon - grey
* `help`: blue shield big icon - grey
* `danger`: red shield big icon - red text, paper background

A `small` boolean prop can be added to reduce font-size and padding/margins, working in addition to the `type` prop.

### Parts of the app affected / Test plan

* New Alert component for all styles:
  - Note: the TopBanner will not be Update Info variant will not be an alert, and stay in TopBanner component
* Disclaimer component (only used in Settings/Experimental) is replaced by the Alert "security"
  - Note: minor changes to margins
* InfoBox component is replaced by the Alert
  - Note: explicitely use type="primary" since default is now "secondary"
  - Used in many places, but since Alert take inspiration from InfoBox, it should not introduce any regression
  - replaced onLearnMore by learnMoreUrl to avoid repetitive useCallbacks and openURL everywhere
* TokenTips is replaced by Alert primary
  - used in Receive and Add account to display info for Ethereum tokens.
  - Note: while the message in receive looks to be a warning, i kept the "primary" styling to avoir regression.
* Tip (in DisclaimerModal - FirmwareUpdate confirmation modal) is replaced by Alert
  - NOTE: a few differences are noticeable. Margins, font-size and line-height are different, the learnMoreLink is on the same line (as specified by the DS), and lack the external icon (~should it be added on Alert somehow ?~ it has been added by default)
   image
* WarnBox component now renders an Alert (type "help) with margin to avoid changing all families code.
* ErrorBanner uses Alert "error" and changes styles:
image
* SpendableBanner, SupplyBanner and WithdrawableBanner (in Lend flows) uses a `small` Alert:
  - NOTE: icon does not change size like in figma
  image
* It also fixes Polkadot warnings out of Design System (uses warning Alerts)

### CODE REVIEW

File: src/renderer/modals/Swap/steps/StepFinished.js [hunk 1]
```diff
@@ -110,6 +110,7 @@ const StepFinished = ({
       <Text p={20} textAlign="center" color="palette.text.shade50" ff="Inter|Regular" fontSize={4}>
         <Trans i18nKey={`swap.modal.steps.finished.description`} />
       </Text>
+      {/* FIXME: transform to Alert */}
       <Disclaimer horizontal p={2} mt={3}>
```
Comment (rjarasson-ledger) [inline src/renderer/modals/Swap/steps/StepFinished.js:114]: I was unable to check the rendering of this component, so i just did not risk to break it.

**Review by valpinkman — DISMISSED** [review]

**Comment by rjarasson-ledger** [comment]
Still work to do:

* rebase after https://github.com/LedgerHQ/ledger-live-desktop/pull/3673 is merged
* change type "tip" to "hint" + resolve conflict due to https://github.com/LedgerHQ/ledger-live-desktop/pull/3562 being merged
* update snapshots

**Review by nabil-brn — DISMISSED** [review]

**Review by valpinkman — DISMISSED** [review]

**Comment by valpinkman** [comment]
@rjarasson-ledger something is broken with the screenshots

**Review by valpinkman — DISMISSED** [review]

**Review by JunichiSugiura — CHANGES_REQUESTED** [review]
Conflicts 🙏

**Comment by gre** [comment]
@JunichiSugiura as maintainers we'll take care of it

**Review by Arnaud97234 — DISMISSED** [review]
I couldn't test all the "Alerts" in LL but at least one of each type. I don't see any bugs so we can merge ✅

**Pull request merged**